docs/design.md

```diff
@@ -195,6 +195,14 @@ Local paths are currently prioritised over other forms. It is presumed
 - Index offsets need to be relative to the CARv1, and not absolute in the
   physical CARv2 file.
 - Index needs to be iterable.
+- Given a CARv2 file, we should be able to decompose it to the corresponding Carv1 payload and the Index.
+- Given a Carv1 file, we need to get it's Index.
+- Given a CARv2 file, we want a ReadOnlyBlockstore on top of it.  
+- Given a CARv1 file and it's Index, we want a ReadOnlyBlockstore wrapper on top if it.
+- We want a ReadWrite blockstore interface that can give us a CarV2 file when we commit. This is important for Graphsync.  
+    Need to flesh out whether we can write arbitary IPLD blocks or whether only coherent DAGs based on (root, selector) combo are allowed.
+- We should be able to read the leading N bytes of a CAR file and determine if it's a CARv1 or a Carv2.
```

Comment:
Do you care about version, or do you care about if it has an index already?

Reply:
If it's a CARv2 you will need to determine whether it has an index or not. Possibly a:

```go
car.Info(io.Reader) (version, Characteristics)
```

Would make sense as a function, or the equivalent on a CAR object if we want it as a method?

Reply:
we might do it as a method on a parsed `Header`, so that you only need to read that once, since you'll be parsing the header to get the `Info` anyway.

so concretely [`ReadHeader(br *bufio.Reader) (*CarHeader, error)`](https://github.com/ipld/go-car/blob/master/car.go#L72) would have a `Version()` that tells you if it's either v1 or v2 that's been read, and would have a `HasIndex() bool` or similar.